brave/brave-core
# Implement Request Off-The-Record

Resolves https://github.com/brave/brave-browser/issues/28750

## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Create a new profile
2. Visit a request-OTR enrolled site (any of these sites will work: https://github.com/brave/adblock-lists/blob/master/brave-lists/request-otr.json)
3. Confirm that the “would you like OTR mode” interstitial appears
4. Choose to go into OTR mode
5. Confirm that the requested page appears, and that there is an info bar at the top of the page describing how to leave OTR mode
6. Open dev tools and go into the console and run this code: `window.localStorage.braveQATest = true`
7. Close the OTR’ed site
8. Confirm that the OTR’ed site is no included in your browsing history
9. revisit the site, this time declining OTR mode
10. open the dev tools console and run this code `console.log(window.localStorage.braveQATest)`
11. confirm that `undefined` is returned
